src/protocols/types/datastore.js

```diff
@@ -830,7 +845,23 @@ class DataStoreReqUpdateInfo extends NEXTypes.Structure {
 	 * @param {Stream} stream NEX data stream
 	 */
 	parse(stream) {
-		this.version = stream.readUInt32LE();
+		let nexVersion;
+		if (stream.connection.title.nex_datastore_version) {
+			nexVersion = stream.connection.title.nex_datastore_version;
+		} else {
+			nexVersion = stream.connection.title.nex_version;
+		}
+
+		if (nexVersion.major >= 3) {
+			this.version = stream.readUInt32LE();
+
+			// * Hack to get the proper type when encoding to JSON
+			this.versionType = 'uint32';
```

Comment:
since `versionType` is not really part of the type, this should probably be made into a private field like `#versionType` to differentiate it at a glance